cluster-operator/src/test/java/io/strimzi/operator/cluster/model/KafkaMirrorMaker2ClusterTest.java

```diff
@@ -0,0 +1,1233 @@
+        assertThat(svc.getSpec().getPorts().get(0).getProtocol(), is("TCP"));
+        assertThat(svc.getSpec().getPorts().get(1).getName(), is(AbstractModel.METRICS_PORT_NAME));
+        assertThat(svc.getSpec().getPorts().get(1).getPort(), is(new Integer(KafkaCluster.METRICS_PORT)));
+        assertThat(svc.getSpec().getPorts().get(1).getProtocol(), is("TCP"));
+        assertThat(svc.getMetadata().getAnnotations(), is(kmm2.getPrometheusAnnotations()));
+
+        checkOwnerReference(kmm2.createOwnerReference(), svc);
+    }
+
+    @Test
+    public void testGenerateServiceWithoutMetrics()   {
+        KafkaMirrorMaker2 resource = new KafkaMirrorMaker2Builder(this.resource)
+                .editSpec()
+                    .withMetrics(null)
+                .endSpec()
+                .build();
+        KafkaMirrorMaker2Cluster kmm2 = KafkaMirrorMaker2Cluster.fromCrd(resource, VERSIONS);
+        Service svc = kmm2.generateService();
+
+        assertThat(svc.getSpec().getType(), is("ClusterIP"));
+        assertThat(svc.getMetadata().getLabels(), is(expectedLabels(kmm2.getServiceName())));
+        assertThat(svc.getSpec().getSelector(), is(expectedSelectorLabels()));
+        assertThat(svc.getSpec().getPorts().size(), is(1));
+        assertThat(svc.getSpec().getPorts().get(0).getPort(), is(new Integer(KafkaMirrorMaker2Cluster.REST_API_PORT)));
+        assertThat(svc.getSpec().getPorts().get(0).getName(), is(KafkaMirrorMaker2Cluster.REST_API_PORT_NAME));
+        assertThat(svc.getSpec().getPorts().get(0).getProtocol(), is("TCP"));
+
+        assertThat(svc.getMetadata().getAnnotations().containsKey("prometheus.io/port"), is(false));
+        assertThat(svc.getMetadata().getAnnotations().containsKey("prometheus.io/scrape"), is(false));
+        assertThat(svc.getMetadata().getAnnotations().containsKey("prometheus.io/path"), is(false));
+
+        checkOwnerReference(kmm2.createOwnerReference(), svc);
+    }
+
+    @Test
+    public void testGenerateDeployment()   {
+        Deployment dep = kmm2.generateDeployment(new HashMap<String, String>(), true, null, null);
+
+        assertThat(dep.getMetadata().getName(), is(KafkaMirrorMaker2Resources.deploymentName(cluster)));
+        assertThat(dep.getMetadata().getNamespace(), is(namespace));
+        Map<String, String> expectedLabels = expectedLabels();
+        assertThat(dep.getMetadata().getLabels(), is(expectedLabels));
+        assertThat(dep.getSpec().getSelector().getMatchLabels(), is(expectedSelectorLabels()));
+        assertThat(dep.getSpec().getReplicas(), is(new Integer(replicas)));
+        assertThat(dep.getSpec().getTemplate().getMetadata().getLabels(), is(expectedLabels));
+        assertThat(dep.getSpec().getTemplate().getSpec().getContainers().size(), is(1));
+        assertThat(dep.getSpec().getTemplate().getSpec().getContainers().get(0).getName(), is(KafkaMirrorMaker2Resources.deploymentName(this.cluster)));
```

Comment:
Please factor out a var for `dep.getSpec().getTemplate().getSpec().getContainers().get(0)` (long lines make code review more painful).

Reply:
Done